### PR TITLE
feat: Moving to new RL layout to be more compatible with external frameworks + models

### PR DESCRIPTION
**Pull request by TomMcL**
### Description

Restructuring RL framework to make it more extensible and flexible. The new layout broadly aims to keep the RL agent entirely separate from concerns of the market sim itself. It does this by introducing the concept of a `puppet` agent and its associated `Action` object. A puppet agent blindly takes an Action agent and performs an action derived from the specification of that action. This allows us to create an RL agent externally to the environment which simply receives a state and outputs one of these Action objects.

In addition, this newly separated setup allows us to integrate with [Gymnasium](https://github.com/Farama-Foundation/Gymnasium) and, eventually, [PettingZoo](https://github.com/Farama-Foundation/PettingZoo). A first draft of an integration with [StableBaselines3](https://stable-baselines3.readthedocs.io/en/master/) has been created and included in the PR. This environment may or may not meet the full Gymnasium custom environment spec, but is sufficient for running predefined RL models from `StableBaselines`. For a simple example of this, see `run.py` within the `stable_baselines` folder.

A later PR will give a fuller description of the steps for adding new reward functions and state descriptions.

### Testing

Manually ran locally to observe learning process. Output files to tensor board for inspection
image

### Breaking Changes

Should be none, RL current setup still in place.

### CODE REVIEW

File: vega_sim/reinforcement/v2/agents/puppets.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+
+class MarketOrderPuppet(Puppet):
+    def initialise(self, vega: VegaService, create_wallet: bool = True):
+        super().initialise(vega, create_wallet)
+        self.market_id = [
+            m.id
+            for m in self.vega.all_markets()
+            if m.tradable_instrument.instrument.name == self.market_name
+        ][0]
```
Comment (cdummett) [inline vega_sim/reinforcement/v2/agents/puppets.py]: Could condense this?
```suggestion
        self.market_id = self.vega.find_market_id(name=self.market_name)
```

File: vega_sim/reinforcement/v2/stable_baselines/environment.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        )[self.learner_name]
+
+        self.current_step = 0
+        return step_res.observation.to_array(), {}
+
+    def render(self, mode="human"):
+        pass
+
+    def close(self):
+        print("CLOSE CALLED")
```
Comment (cdummett) [inline vega_sim/reinforcement/v2/stable_baselines/environment.py]: Switch to logging or just remove?

File: vega_sim/reinforcement/v2/states.py [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+#             ),
+#             bid_prices=bid_prices,
+#             ask_prices=ask_prices,
+#             bid_volumes=[level.volume for level in book_state.buys]
+#             + [0] * max(0, self.num_levels - len(book_state.buys)),
+#             ask_volumes=[level.volume for level in book_state.sells]
+#             + [0] * max(0, self.num_levels - len(book_state.sells)),
+#             trading_fee=fee,
+#             next_price=next_price,
+#         )
```
Comment (cdummett) [inline vega_sim/reinforcement/v2/states.py]: Is this needed?

**Review by cdummett — DISMISSED** [review]
Looks really cool! a few small comments

File: vega_sim/reinforcement/v2/stable_baselines/environment.py (hunk 2 above)
Comment (TomMcL) [reply vega_sim/reinforcement/v2/stable_baselines/environment.py]: Yeah that was meant to die before I pushed :D

File: vega_sim/reinforcement/v2/states.py (hunk 3 above)
Comment (TomMcL) [reply vega_sim/reinforcement/v2/states.py]: ❌

**Review by cdummett — APPROVED** [review]
LGTM

**Pull request merged**